doc/theory.md

```diff
@@ -52,3 +52,18 @@ Finally, FLOSS diffs the emulator memory states from before and
   6. Emulate decoder functions using extracted arguments and emulator state
   7. Diff memory state from before and after decoder emulation
   8. Extract human-readable strings from memory state difference
+
+## Go String Extraction
+The Go string extraction algorithm implemented in the FLOSS tool is designed to address the unique characteristics of Go strings in binaries, ensuring efficient retrieval during reverse engineering and malware analysis tasks. Go strings are conventionally represented as sequences of 8-bit bytes, often encoding UTF-8 text. Without special handling, the strings within the binary may appear as a large chunk of indistinguishable data. The FLOSS algorithm overcomes this challenge by analyzing struct string instances and leveraging the length-sorted order of strings within the binary. This capability provides valuable insights into the inner workings of Go programs, facilitating comprehensive analysis and understanding.
```

Comment:
```suggestion
Programs compiled by the Go compiler use a string representation that is difficult to interpret by humans. Although they are UTF-8 encoded, and therefore show up in the output of `strings.exe`, program strings are not NULL-terminated. This means separate strings within the binary may appear as a large chunk of indistinguishable string data.

FLOSS implements an algorithm to handle the unusual characteristics of strings in Go binaries. This approach analyzes instances of the `struct String` type to identify candidate strings and reasons about the length-sorted order to avoid false positives. Crucially, FLOSS automatically handles the complexities of Go strings and displays strings as written in the program's source code.
```

Reply:
we may also want to include text about other sorts of strings, like runtime strings, which are not derived from the program strings.

Reply:
also we may want to note the fact that there are some optimizations, like inlining constants, that we don't handle well yet.

Reply:
should add links to the Go project and to the source code of the `struct String` layout. maybe also a link to a place that interested readers can learn more.